test/test_fx.py

```diff
@@ -1263,5 +1268,28 @@ def forward(self, x: List[str]) -> List[str]:
         scripted = torch.jit.script(traced)
         self.assertIn("-> List[str]", scripted.code)
 
+    def test_user_friendly_call_provenance_with_function(self):
+        def fn(x):
+            return wrapper_fn(x)
+
+        traced = torch.fx.symbolic_trace(fn)
+
+        with self.assertRaisesRegex(RuntimeError, "'wrapper_fn' is "
+                                    "being compiled since it was called"
+                                    " from 'function.forward'"):
```

Comment:
The user may be confused that we're saying `<fn_name>.forward` but not sure of a quick solution to this, since we're promoting functions to Modules in symbolic tracing

"""All abstractions are leaky abstractions when it comes to error reporting" - Me" - Wayne Gretzky" - Michael Scott

Reply:
Same here--agree that it's confusing, not sure what to do about it. An earlier version of my code set the module name to `nnModule_with_autogenerated_forward_wrapping_user_function_FUNCTION-NAME`, which is clearer, but also way too long. I'll keep it as-is unless you think renaming the module would be better.